tests/network/mod.rs

```diff
@@ -27,6 +27,12 @@ impl<D: DistAlgorithm> TestNode<D> {
         &self.outputs
     }
 
+    /// Returns whether the algorithm has terminated.
+    #[allow(unused)] // Not used in all tests.
```

Comment:
So, your preference is to define methods as opposed to `pub` fields?

Reply:
In general I'd tend towards yes: Encapsulate more, expose less. In this particular case I just thought it was nice to be able to write `nodes.values().all(TestNode::terminated)`. If you prefer, I'm happy to revert this.